sdk/cosmos/azure-cosmos/azure/cosmos/_base.py

```diff
@@ -334,7 +335,7 @@ def IsNameBased(link):
     parts = link.split("/")
 
     # First part should be "dbs"
-    if len(parts) == 0 or not parts[0] or not parts[0].lower() == "dbs":
+    if not (parts and parts[0].lower() == "dbs"):
```

Comment:
After transforming `len(parts) == 0` to `not parts` the clause has the form `not X or not Y or not Z` which is equivalent to `not (X and Y and Z)` (deMorgan's law). Additionally, the check for `parts[0]` is not needed -- if parts has any elements, it has `parts[0]` and `parts[0]` is a string (output of `split`)